base/server/scripts/operations

```diff
@@ -804,10 +804,24 @@ backup_instance_configuration_files()
     return 0
 }
 
+check_configuration() {
+
+    echo "Checking for deprecated algorithms"
+
+    # Only check PKI server configuration files.
+    # Exclude NSS database files and links to Tomcat standard configuration files.
+    # The result will appear in systemd journal when the server is started.
+
+    FILES=$(find /etc/pki/${pki_instance_id} -type f -not -path "/etc/pki/${pki_instance_id}/alias/*")
+    grep -E "SHA1|SHA-1|SHA_1" $FILES
```

Comment:
This regex won't pick up `SHA` which is an alias for `SHA-1`. I appreciate that complicates things massively.

Reply:
Yeah, if we want to be precise we'll need to handle each configuration parameter individually, which would require a lot of work to implement. I think this static config check with the runtime check you just implemented should be sufficient for now.

Reply:
Or how about this?
```
grep -E "SHA$|SHA,|SHAwith|SHA1$|SHA1,|SHA1with|SHA-1$|SHA-1,|SHA_1$|SHA_1," $FILES
````
Do we need to add `SHA-1with` and `SHA_1with` too? Are there other variants that we need to add?

Reply:
@jmagne @ladycfu Could you take a look at this? ^^

Reply:
I think you can do better with the regex `^SHA([^2-9]{0,2})$`. Testing on https://www.regexpal.com/
```
SHA (hit)
SHA1 (hit)
SHA-1 (hit)
SHA_1 (hit)
SHA256 (miss)
SHA-256 (miss)
SHA_256 (miss)
```

Reply:
Hmm.. that probably wouldn't work on a comma-separated list of algs.

Reply:
It works if you drop the start/end markers so `SHA([^2-9]{0,2})`. I tried at the link before with the test string:
```
policyset.adminCertSet.8.constraint.params.signingAlgsAllowed=SHA256withRSA,SHA512withRSA,SHAwithDSA,SHA1withEC,SHA256withEC,SHA384withRSA,SHA384withEC,SHA512withEC,SHA256withRSA/PSS,SHA384withRSA/PSS,SHA512withRSA/PSS
```
It hits at `SHAwithDSA`, which is right. If you remove that it hits `SHA1withEC`. If you remove that it misses the string.

Reply:
It won't find every instance though, but maybe you could fix it so it does?

Reply:
Without the start/end markers it would match `SHA384withRSA`.

Reply:
Oh sorry it should have been `SHA([^2-9]{1,2})`, then it only matches `SHA` and the various permutations of `SHA-1`, and doesn't hit the others. I was very confused for a second there :smile: